src/pages/buy-sell/components/CurrencyDropdown/CurrencyDropdown.tsx

```diff
@@ -19,6 +19,15 @@ const CurrencyDropdown = ({ selectedCurrency, setSelectedCurrency }: TCurrencyDr
     const { data } = api.settings.useSettings();
     const { isDesktop, isMobile } = useDevice();
     const [showCurrencySelector, setShowCurrencySelector] = useState<boolean>(false);
+    const { deleteQueryString, setQueryString } = useQueryString();
+
+    const onToggleCurrencyDropdown = (shouldShow: boolean) => {
+        setShowCurrencySelector(shouldShow);
+        if (!isDesktop) {
+            if (shouldShow) setQueryString({ modal: 'CurrencyFilterModal' });
+            else deleteQueryString('modal');
+        }
+    };
```

Comment:
reason why i added this queryString to modal directly is to ensure we hide the banner if currency selector is shown. And in the index.tsx file, I have some logic that handles showing the banner if its either the RadioGroupModal or FundsModal or if there is no Modal showing at all. Below you can see the issue

https://github.com/user-attachments/assets/7d7454f8-441a-41be-bd3e-2df8f658c3c4

https://github.com/user-attachments/assets/59ba55de-cf47-4e02-b26c-5bea68d788b0

Reply:
also you cannot use isModalOpenFor check because it only returns true in whatever component you have called showModal in